src/GafferScene/CollectPrimitiveVariables.cpp

```diff
@@ -0,0 +1,259 @@
+using namespace IECore;
+using namespace Gaffer;
+using namespace GafferScene;
+
+//////////////////////////////////////////////////////////////////////////
+// Internal utilities
+//////////////////////////////////////////////////////////////////////////
+
+namespace
+{
+
+static InternedString g_collectPrimitiveVariablesSuffix( "collectPrimitiveVariables:suffix" );
+
+} // namespace
+
+//////////////////////////////////////////////////////////////////////////
+// CollectPrimitiveVariables
+//////////////////////////////////////////////////////////////////////////
+
+IE_CORE_DEFINERUNTIMETYPED( CollectPrimitiveVariables );
+
+size_t CollectPrimitiveVariables::g_firstPlugIndex = 0;
+
+CollectPrimitiveVariables::CollectPrimitiveVariables( const std::string &name )
+	:	SceneElementProcessor( name, IECore::PathMatcher::NoMatch )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+
+	addChild( new StringVectorDataPlug( "suffixes", Plug::In, new StringVectorData() ) );
+	addChild( new StringPlug( "targetVariables", Plug::In, "P" ) );
+	addChild( new BoolPlug( "isVariation", Plug::Out, false ) );
+
+	// Fast pass-throughs for things we don't modify
+	outPlug()->attributesPlug()->setInput( inPlug()->attributesPlug() );
+	outPlug()->transformPlug()->setInput( inPlug()->transformPlug() );
+	outPlug()->boundPlug()->setInput( inPlug()->boundPlug() );
+}
+
+CollectPrimitiveVariables::~CollectPrimitiveVariables()
+{
+}
+
+Gaffer::StringVectorDataPlug *CollectPrimitiveVariables::suffixesPlug()
+{
+	return getChild<StringVectorDataPlug>( g_firstPlugIndex + 0 );
+}
+
+const Gaffer::StringVectorDataPlug *CollectPrimitiveVariables::suffixesPlug() const
+{
+	return getChild<StringVectorDataPlug>( g_firstPlugIndex + 0 );
+}
+
+Gaffer::StringPlug *CollectPrimitiveVariables::targetVariablesPlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 1 );
+}
+
+const Gaffer::StringPlug *CollectPrimitiveVariables::targetVariablesPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 1 );
+}
+
+Gaffer::BoolPlug *CollectPrimitiveVariables::isVariationPlug()
+{
+	return getChild<BoolPlug>( g_firstPlugIndex + 2 );
+}
+
+const Gaffer::BoolPlug *CollectPrimitiveVariables::isVariationPlug() const
+{
+	return getChild<BoolPlug>( g_firstPlugIndex + 2 );
+}
+
+void CollectPrimitiveVariables::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	SceneElementProcessor::affects( input, outputs );
+
+	if(
+		input == inPlug()->objectPlug() ||
+		input == suffixesPlug() ||
+		input == targetVariablesPlug()
+	)
+	{
+		outputs.push_back( isVariationPlug() );
+		outputs.push_back( outPlug()->objectPlug() );
+	}
+}
+
+void CollectPrimitiveVariables::hash( const Gaffer::ValuePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	SceneElementProcessor::hash( output, context, h );
+	if( output == isVariationPlug() )
+	{
+		if( !( filterValue( context ) & IECore::PathMatcher::ExactMatch ) )
+		{
+			h = IECore::MurmurHash();
```

Comment:
What's the intention here? A hash always needs to [start with a TypeId](https://github.com/GafferHQ/gaffer/blob/master/src/Gaffer/ComputeNode.cpp#L57), so I think the right thing to do is just return, using the basic hash computed by the call to `SceneElementProcessor::hash()`.

Reply:
Ah, OK.  I wasn't sure what the right thing to do here is.  At this point, I know that the value is always going to be false, so it seemed like it would make sense to return a fixed hash, but yeah, the convention makes sense.